test/versioned/undici/requests.tap.js

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2021 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const tap = require('tap')
+const { DESTINATIONS } = require('../../../lib/config/attribute-filter')
+const helper = require('../../lib/agent_helper')
+const metrics = require('../../lib/metrics_helper')
+
+tap.test('Undici request tests', (t) => {
+  t.autoend()
+
+  let agent
+  let undici
+
+  t.before(() => {
+    agent = helper.instrumentMockedAgent({
+      feature_flag: {
+        undici_instrumentation: true
+      }
+    })
+
+    undici = require('undici')
+  })
+
+  t.teardown(() => {
+    helper.unloadAgent(agent)
+  })
+
+  t.test('should not fail if request not in a transaction', async (t) => {
+    const { statusCode } = await undici.request('https://httpbin.org', {
+      path: '/post',
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application.json'
+      },
+      body: Buffer.from(`{"key":"value"}`)
+    })
+
+    t.equal(statusCode, 200)
+    t.end()
+  })
+
+  t.test('should properly name segments', (t) => {
+    helper.runInTransaction(agent, async (tx) => {
+      const { statusCode } = await undici.request('https://httpbin.org', {
+        path: '/post',
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application.json'
+        },
+        body: Buffer.from(`{"key":"value"}`)
+      })
+      t.equal(statusCode, 200)
+
+      metrics.assertSegments(tx.trace.root, ['External/httpbin.org/post'], { exact: false })
+      tx.end()
+      t.end()
+    })
+  })
+
+  t.test('should add attributes to external segment', (t) => {
+    helper.runInTransaction(agent, async (tx) => {
+      const { statusCode } = await undici.request('https://httpbin.org', {
+        path: '/get?a=b&c=d',
+        method: 'GET'
+      })
+      t.equal(statusCode, 200)
+      const segment = metrics.findSegment(tx.trace.root, 'External/httpbin.org/get')
+      const attrs = segment.getAttributes()
+      t.equal(attrs.url, 'https://httpbin.org/get')
+      t.equal(attrs.procedure, 'GET')
+      const spanAttrs = segment.attributes.get(DESTINATIONS.SPAN_EVENT)
+      t.equal(spanAttrs['http.statusCode'], 200)
+      t.equal(spanAttrs['http.statusText'], 'OK')
+      t.equal(spanAttrs['request.parameters.a'], 'b')
+      t.equal(spanAttrs['request.parameters.c'], 'd')
```

Comment:
i wouldn't expect to see these without adding the attributes to the include list. i think we have a bug here and with the http-outbound instrumentation. will have to double check the specs / attribute in context work though to see if we handle different for spans specifically for some reason.

Reply:
let me know what you want to do here.  we talked in slack and I'm not sure if I should address in here or keep and open different PR as this also affects http-outbound

Reply:
I'd tackle in a different PR. The functionality has already landed (behind feature flag).